### ISSUE TITLE
fix area bounds in non-monotonic coordinates

### ISSUE THREAD

**Issue by jmilloy** [creare-com/podpac#162]
**Description**
The area bounds of non-monotonic coordinates is incorrect when ctype is not 'point'.

**Steps to Reproduce**
```
c = ArrayCoordinates1d([20., 50., 60., 90., 40., 10.])
c.area_bounds
```

**Observed Behavior**
```
>>> c.area_bounds
array([ 5., -5.])
```

**Expected Behavior**
Either
 * the `area_bounds` should be inferred reasonably from the coordinates, or
 * an exception should be raised

**Additional Notes**
The area_bounds also works correctly when using ctype 'point`. For segment ctypes, the area_bounds works correctly when the `extents` are provided. It may not be possible to reasonably infer the area_bounds from non-monotonic coordinates, so the solution may be to require extents when defining  non-monotonic segment coordinates.

Alternatively, a reasonable inference for area_bounds would be to sort the coordinates and use the same value we would use for monotonic coordinates. In the example case, that would be `array([  5., 105.])`.

**Comment by jmilloy** [comment]
See also #120

**Comment by jmilloy** [comment]
@mpu-creare will you let me know if the inference or the exception is better? Both are quick to implement.

**Comment by mpu-creare** [comment]
Please do the inference. 
[min - segment_length_of_min / 2, max + segment_length_of_max / 2] for not point
[min , max ] for point

If the user does not specify `segment_length` for ctype `not 'point'`, raise an exception with a clear message. 

Thanks!

**Comment by jmilloy** [comment]
Okay, so this depends on #120.

**Comment by jmilloy** [comment]
Implemented in #163
